File: arrays/easy/is_subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int p1 = 0;
        int p2 = 0;
        int l1 = s.length();
        int l2 = t.length();

        while(p1 < l1 && p2 < l2){
            if(s[p1] == t[p2]){
                p1++;
            }
            p2++;
        }
        return p1==l1;
    }
};

File: strings/easy/is_subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:

        p1 = p2 = 0
        n1 = len(s)
        n2 = len(t)
        while p1 < n1 and p2 < n2:
            if s[p1] == t[p2]:
                p1 += 1
            p2 += 1
        return p1 == n1
